lib/cfg-lex.l

```diff
@@ -255,6 +255,8 @@ word	[^ \#'"\(\)\{\}\[\]\\;\r\n\t,|\.@:]
      /* rules to parse a block as a LL_BLOCK token */
 <block>{white}+            ;
 <block>[^{( \t]+           { return LL_ERROR; }
+<block>\({white}*\)        { yy_pop_state(yyscanner); yylval->cptr = NULL; return LL_BLOCK; }
```

Comment:
I'm not an expert of lexer. Do we need pop_state here?
Can it cause an issue if in block context/state we pop the state, but in the next line we find an option with default value?
I've tried to test it but so far no problem found.

Reply:
Yes, that pop state is needed. Although somewhat not intuitive because the push state is triggered from the grammar, and one could argue that the pop could be moved there for the same reason. (Not directly in the grammar, but triggered by an item pushed by the grammar) But we need to pop the block state at this point, because the next item going to be an `LL_INDENTIFIER`  (if config is valid), that is not part of the block.

No it cannot.  There is a general rule `<*>\r?\n` that takes care of the new line, so this is a reason it is not a problem.

Reply:
Thanks for clarifying!